src/MagnumPlugins/ObjImporter/Test/multiMaterial.mtl

```diff
@@ -0,0 +1,8 @@
+newmtl mat_0
+Ka 1 1 1
+map_Ka Textures/mat_0.tga
+
+
+newmtl mat_1
+Ka 1 1 1
+map_Ka Textures/mat_1.tga
```

Comment:
I think this problem would go away completely with the single-pass parser design I outlined above -- if there would be no newline at the end, the parser loop would just not ignore anything at the end and then not go into another loop iteration.